web/src/hooks/useApps.ts

```diff
@@ -301,19 +312,46 @@ const useApps = () => {
     [insertNewAppMutation]
   );
 
+  const parseDescription = (currentApp: AppModel | null) => {
+    if (currentApp && currentApp.description_internal) {
+      try {
+        return JSON.parse(currentApp.description_internal);
+      } catch (error) {
+        console.error("Failed to parse description_internal:", error);
+        return {
+          description_overview: currentApp.description_internal,
+          description_how_it_works: "",
+          description_connect: "",
+        };
+      }
+    }
+    return {};
+  };
+
+  const encodeDescription = (
+    description_overview: string,
+    description_how_it_works: string = "",
+    description_connect: string = ""
+  ) => {
+    return JSON.stringify({
+      [descriptionSubFields[0]]: description_overview,
```

Comment:
@andy-t-wang I suggest to make a TS enum here instead of descriptionSubFields object.